src/redis_lock/__init__.py

```diff
@@ -156,7 +157,9 @@ def __init__(self, redis_client, name, expire=None, id=None, auto_renewal=False)
         self._held = id is not None
         self._name = 'lock:'+name
         self._signal = 'lock-signal:'+name
-        self._lock_renewal_interval = expire*2/3 if auto_renewal else None
+        self._lock_renewal_interval = (float(expire)*2/3
+                                       if auto_renewal
+                                       else None)
```

Comment:
I think the problem was, that a thread holds a reference to a lock while extending and if an interval is zero andd PyPy is used, considering, that in PyPy objects are not destroyed immediately after garbage collection, lock might been referenced forever.